src/backup.py

```diff
@@ -128,3 +161,255 @@ def can_use_bucket(self) -> bool:
             )
             return False
         return True
+
+
+def create_backup(
+    container: ops.Container,
+    s3_parameters: S3Parameters,
+    passphrase: str,
+    backup_id: Optional[str] = None,
+) -> str:
+    """Create a backup for Synapse running it in the workload.
+
+    Args:
+        container: Synapse Container
+        s3_parameters: S3 parameters for the backup.
+        passphrase: Passphrase use to encrypt the backup.
+        backup_id: Name of the object in the backup.
+           It will be autogenerated if it is not set.
+
+    Returns:
+       The backup key used for the backup.
+
+    Raises:
+       BackupError: If there was an error creating the backup.
+    """
+    if backup_id is None:
+        backup_id = "backup-" + datetime.datetime.now().strftime(BACKUP_ID_FORMAT)
+
+    _prepare_container(container, s3_parameters, passphrase)
+    paths_to_backup = _get_paths_to_backup(container)
+    logger.info("Paths to back up: %s.", list(paths_to_backup))
+    if not paths_to_backup:
```

Comment:
Is it unreasonable to think there might be no files to backup? Perhaps it would be better to log a warning instead of raising an exception here or even silently return. Is there any issue if the backup is executed passing an empty list of paths?

Reply:
I understand that this should never happen. As Synapse should have started, at least the keys should be there... it should be an error